source/Locales/Compactness/Properties.lagda

```diff
@@ -0,0 +1,716 @@
+---
+title:          Properties of compactness
+author:         Ayberk Tosun
+date-started:   2024-07-19
+date-completed: 2024-07-31
+---
+
+We collect properties related to compactness in locale theory in this module.
+This includes the equivalences to two alternative definitions of the notion of
+compactness, which we denote `is-compact-open'` and `is-compact-open''`.
+
+\begin{code}[hide]
+
+{-# OPTIONS --safe --without-K #-}
+
+open import MLTT.Spartan hiding (J)
+open import UF.Base
+open import UF.Classifiers
+open import UF.FunExt
+open import UF.PropTrunc
+open import UF.Sets
+open import UF.Subsingletons
+open import UF.Subsingletons-FunExt
+open import UF.Subsingletons-Properties
+open import UF.SubtypeClassifier
+
+module Locales.Compactness.Properties
+        (pt : propositional-truncations-exist)
+        (fe : Fun-Ext)
+        (pe : Prop-Ext)
+       where
+
+open import Fin.Kuratowski pt
+open import Fin.Type
+open import Locales.Compactness.Definition pt fe
+open import Locales.Frame pt fe
+open import Locales.WayBelowRelation.Definition  pt fe
+open import MLTT.List using (member; []; _∷_; List; in-head; in-tail; length)
+open import Slice.Family
+open import Taboos.FiniteSubsetTaboo pt fe
+open import UF.Equiv hiding (_■)
+open import UF.EquivalenceExamples
+open import UF.ImageAndSurjection pt
+open import UF.Logic
+open import UF.Powerset-Fin pt hiding (⟨_⟩)
+open import UF.Powerset-MultiUniverse
+open import UF.Sets-Properties
+
+open AllCombinators pt fe
+open Locale
+open PropositionalTruncation pt
+
+\end{code}
+
+\section{Preliminaries}
+
+Given a family `S`, we denote the type of its subfamilies by `SubFam S`.
+
+\begin{code}
+
+SubFam : {A : 𝓤  ̇} {𝓦 : Universe} → Fam 𝓦 A → 𝓦 ⁺  ̇
+SubFam {𝓤} {A} {𝓦} (I , α) = Σ J ꞉ 𝓦  ̇ , (J → I)
+
+\end{code}
+
+Tiny lemma recording the fact that nothing is a member of the empty list.
+
+\begin{code}
+
+not-in-empty-list : {A : 𝓤  ̇} {x : A} → ¬ ∥ member x [] ∥
+not-in-empty-list = ∥∥-rec 𝟘-is-prop (λ ())
+
+\end{code}
+
+Given any list, the type of elements that fall in the list is a
+Kuratowski-finite type.
+
+\begin{code}
+
+nth : {X : 𝓤  ̇} → (xs : List X) → (i : Fin (length xs)) → Σ x ꞉ X , ∥ member x xs ∥
+nth         (x ∷ _)  (inr ⋆) = x , ∣ in-head ∣
+nth {_} {X} (_ ∷ xs) (inl n) = x , ∥∥-functor in-tail (pr₂ IH)
+ where
+  IH : Σ x ꞉ X , ∥ member x xs ∥
+  IH = nth xs n
+
+  x : X
+  x = pr₁ IH
+
+nth-is-surjection : {X : 𝓤  ̇} (xs : List X) → is-surjection (nth xs)
+nth-is-surjection []       (y , μ) = ∥∥-rec ∃-is-prop (λ ()) μ
```

Comment:
Again.

Reply:
There is a problem with moving this into `MLTT.List`: it uses `MLTT.Fin` which imports `MLTT.List` so it results in a cyclic dependency. I’m not sure what the best solution is. Should I create a new module?

Reply:
Probably the best solution is to create MLTT.List-Properties.

Reply:
I have now created a new module `MLTT.List-Properties` and moved these functions there.